libdbus-sys/build_vendored.rs

```diff
@@ -0,0 +1,389 @@
+    "-Wduplicated-branches",
+    "-Wduplicated-cond",
+    "-Wfloat-equal",
+    "-Wformat-nonliteral",
+    "-Wformat-security",
+    "-Wformat=2",
+    "-Winit-self",
+    "-Winline",
+    "-Wlogical-op",
+    "-Wmissing-declarations",
+    "-Wmissing-format-attribute",
+    "-Wmissing-include-dirs",
+    "-Wmissing-noreturn",
+    "-Wnull-dereference",
+    "-Wpacked",
+    "-Wpointer-arith",
+    "-Wredundant-decls",
+    "-Wrestrict",
+    "-Wreturn-type",
+    "-Wshadow",
+    "-Wstrict-aliasing",
+    "-Wswitch-default",
+    "-Wswitch-enum",
+    "-Wundef",
+    "-Wunused-but-set-variable",
+    "-Wwrite-strings",
+
+    // Extra warnings just for C
+    "-Wdeclaration-after-statement",
+    "-Wimplicit-function-declaration",
+    "-Wjump-misses-init",
+    "-Wmissing-prototypes",
+    "-Wnested-externs",
+    "-Wold-style-definition",
+    "-Wpointer-sign",
+    "-Wstrict-prototypes",
+];
+
+/// Wrapper type for the libdbus config.
+///
+/// Boils down to a set of C defs/undefs.
+struct Config<'a> {
+    defs: HashMap<&'a str, Option<String>>,
+    undefs: Vec<&'a str>,
+}
+
+impl<'a> Config<'a> {
+    fn new() -> Self {
+        Self {
+            defs: HashMap::new(),
+            undefs: Vec::new(),
+        }
+    }
+
+    fn enable(&mut self, flag: &'a str) {
+        self.defs.insert(flag, None);
+    }
+
+    fn disable(&mut self, flag: &'a str) {
+        self.undefs.push(flag);
+    }
+
+    fn set<T: ToString>(&mut self, flag: &'a str, value: T) {
+        self.defs.insert(flag, Some(value.to_string()));
+    }
+
+    /// Iterator over all C defines, set with -Dopt=val
+    fn defs<'b>(&'b self) -> Iter<'b, &'a str, Option<String>> {
+        self.defs.iter()
+    }
+
+    /// Iterator over all undefs, that must be set with -Uopt
+    fn undefs<'b>(&'b self) -> std::slice::Iter<'b, &'a str> {
+        self.undefs.iter()
+    }
+}
+
+/// Config.h and dbus-arch-deps.h stubs (we generate config programatically
+/// instead to be passed directly to cc)
+fn generate_stubs() -> Result<(), Box<dyn Error>> {
+    let mut config = File::create("vendor/dbus/config.h")?;
+    let mut arch = File::create("vendor/dbus/dbus/dbus-arch-deps.h")?;
+
+    // Stub out the file.
+    config.write_all(b"
+        #pragma once
+        #ifdef __GNUC__
+        #include <features.h>
+        #if __GNUC_PREREQ(5,0)
+            #pragma GCC diagnostic ignored \"-Wunused-but-set-variable\"
+        #elif __GNUC_PREREQ(8,0)
+            #pragma GCC diagnostic ignored \"-Wduplicated-branches\"
+            #pragma GCC diagnostic ignored \"-Wsign-compare\"
+        #endif
+        #endif
+    ")?;
+
+    // Define integer types.
+    //
+    // The libdbus builder relies on detecting int/long/long long width.
+    // but we can just use modern C types like uint64_t.
+    //
+    // Pointer width will still be detected at build time by our
+    // generate_config() method.
+    arch.write_all(b" 
+        #ifndef DBUS_ARCH_DEPS_H
```

Comment:
Needs reference (if there is one?)

Reply:
Also no static reference, when building with `cmake/meson` this is created. And really all it does is setup the types for `dbus_int16_t`, `dbus_int32_t`, and `dbus_int64_t`. Plus their unsigned counterparts.